graphitron-codegen-parent/graphitron-java-codegen/src/main/java/no/sikt/graphitron/definitions/helpers/ServiceWrapper.java

```diff
@@ -5,12 +5,13 @@
 import no.sikt.graphql.directives.GenerationDirectiveParam;
 
 import static no.sikt.graphql.directives.GenerationDirective.SERVICE;
+import static no.sikt.graphql.directives.GenerationDirective.TABLE_METHOD;
 
 /**
  * Class that contains extended information about a service.
  */
 public class ServiceWrapper extends CodeReferenceWrapper {
     public <T extends NamedNode<T> & DirectivesContainer<T>> ServiceWrapper(T field) {
-        super(field, SERVICE, GenerationDirectiveParam.SERVICE);
+        super(field, field.hasDirective(SERVICE.getName()) ? SERVICE : TABLE_METHOD, GenerationDirectiveParam.SERVICE);
```

Comment:
Her er det litt rart og ved direktiv parameter bytte
Og det er kanskje allerede uharmonisk

Av en grunn så tenkte vi det ble mye ekstra kode vi måtte skrive ved å lage en egen TableMethodWrapper, men kanskje det ikke er så tungvint å gjøre det når jeg ser det med nye øyne. Så jeg tror jeg tester det for å rydde opp dette her samtidig :eyes:

Reply:
Jeg kan jo også gjøre noe triksing som dette:

```suggestion
        super(field, getDirective(field), getDirectiveParam(field));
    }

    private static <T extends NamedNode<T> & DirectivesContainer<T>> no.sikt.graphql.directives.GenerationDirective getDirective(T field) {
        return field.hasDirective(SERVICE.getName()) ? SERVICE : TABLE_METHOD;
    }

    private static <T extends NamedNode<T> & DirectivesContainer<T>> no.sikt.graphql.directives.GenerationDirectiveParam getDirectiveParam(T field) {
        return field.hasDirective(SERVICE.getName()) ? GenerationDirectiveParam.SERVICE : GenerationDirectiveParam.TABLE_REFERENCE;
    }

```

men det føles ikke helt riktig

Reply:
Hvorfor ikke dette?
```
super(field, field.hasDirective(SERVICE.getName()) ? SERVICE : TABLE_METHOD, field.hasDirective(SERVICE.getName()) ? GenerationDirectiveParam.SERVICE : GenerationDirectiveParam.TABLE_REFERENCE);
```

Kan være jeg ikke ser et annet problem her, men noe ekvivalent med det burde vel holde?